uwblueprint/paramedics-react
# Add CCP Dashboard Screen 1

## Overview
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

[Notion task](https://www.notion.so/uwblueprintexecs/CCP-Dashboard-Screen-1-948dac5567794db2bdbc91f444c4af78)
Notes: 
1. Notion task doesn't have updated design, should look at [figma](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-(S20)?node-id=1%3A30)
2. The app bar in the design has not been implemented (it's part of a separate PR)
3. Patient/hospital association has not been implemented in the backend so hospital related contents are fake

## Changes
- Add CCP Dashboard at route `/events/:eventId/ccps/:ccpId`
    - Contents of Patient Overview tab
    - Implement filtering for patient table
- Add common `LoadingState` component

## Testing
- Add some patients to a specific CCP with varying triage levels and statuses
- Navigate to `/events/:eventId/ccps/:ccpId` where eventId and ccpId are valid (ccpId must be associated with the eventId)
- See the loading state when patients are not loaded
- Clicking on any table row in the patient info table should open a dialog with the patient's details
- Try sorting the table rows by any column (default is descending Time)
- Play around with the filters popover:
    1. "Clear Filter" button unchecks all boxes, but does not apply until user clicks "Apply Filter"
    2. Closing the popover (via the "X" icon or by clicking outside of the popover) will unsave the state of checked/unchecked checkboxes
    3. Selected filters (via checkboxes) are not applied until user clicks "Apply Filter" button
        - When no checkboxes are selected for a particular category, it means the patients are not filtered by that category
    4. Chips appear that match the selected AND applied filters
    5. Clicking "X" on a chip will remove that filter (also check that the state of the checkboxes in the popover will reflect this)

## Screenshots
Loading state:
![image](https://user-images.githubusercontent.com/22457859/87224750-c308ed00-c355-11ea-9087-10e597a8f1e2.png)

Loaded CCP Dashboard:
![image](https://user-images.githubusercontent.com/22457859/87217449-5f5fcf00-c317-11ea-8a12-55c475377bf0.png)
![image](https://user-images.githubusercontent.com/22457859/87217635-5ec83800-c319-11ea-9697-695f0fa32f28.png)

Details Dialog:
![image](https://user-images.githubusercontent.com/22457859/87217999-fd09cd00-c31c-11ea-855d-8351de2ec78d.png)

Filters popover:
![image](https://user-images.githubusercontent.com/22457859/87217637-6e478100-c319-11ea-87f5-43324b2d1eea.png)
![image](https://user-images.githubusercontent.com/22457859/87217651-9f27b600-c319-11ea-9274-345ed93614b6.png)
![image](https://user-images.githubusercontent.com/22457859/87217658-a949b480-c319-11ea-83e2-6df3357da189.png)
